### PR TITLE
CDAP-2170 Fetch run record for a suspened program

### PR DESCRIPTION
**Pull request by poornachandra**
Also, fixed getting suspended run records for a program.

JIRA - https://issues.cask.co/browse/CDAP-2170

### CODE REVIEW

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-RBT247-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java [hunk 1]
```diff
@@ -264,9 +263,26 @@ public void recordProgramStop(Id.Program program, String pid, long stopTs, Progr
...
+    RunRecord complete = getCompletedRun(program, runid);
+    if (complete != null) {
+      return complete;
+    }
+
+    // Else query suspended run records
+    return getNonCompleteRun(program, TYPE_RUN_RECORD_SUSPENDED, runid);
+  }
+
+  private RunRecord getNonCompleteRun(Id.Program program, String recordType, String runid) {
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java]: Should we add one line comment saying what is included in the `NonComplete` run?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java [hunk 2]
```diff
@@ -264,9 +263,26 @@ public void recordProgramStop(Id.Program program, String pid, long stopTs, Progr
   // Any changes made here will have to be made over there too.
   // JIRA https://issues.cask.co/browse/CDAP-2172
   public RunRecord getRun(Id.Program program, final String runid) {
-    // For querying running records
+    // Query active run record first
+    RunRecord running = getNonCompleteRun(program, TYPE_RUN_RECORD_STARTED, runid);
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java]: may be rename the method to `getUnfinishedRun`

**Comment by sagarkapare** [comment]
minor comments and LGTM :+1:

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-RBT247-2

**Comment by sreevatsanraman** [comment]
Build passed. So merging.

**Pull request merged**